### PR TITLE
Damola inventory

### PR DESCRIPTION
**Pull request by darmhoo**
#### What does this PR do?
- (replace me) Explain WHY this PR is required. E.g.: What feature it adds, what problem it solves, …

#### Description of Task proposed in this pull request?
- (replace me) A description of WHAT changed in the codebase. Additionally add the reasoning if it's complex or abstract. 

#### How should this be manually tested (Quality Assurance)?
- Explain how to manually test this feature. For example if changes were made in
the UI or in the API, explain where and if any specific access is needed.

#### What are the relevant pivotal tracker stories?
- [Story ID:  replace text with story id](paste Pivotal Tracker link here)

#### Any background context you want to add (Operations Impact)?
- (replace me) List the steps that must be taken for this PR to work.
E.g.: `npm install`, `Add "github_key" to environment variables, ...`

Be sure to include all systems that needs to be changed or which system is
affected by the change (Ex: Requires Styled Components to be installed and
configured in `babel.rc`, file to be updated).

#### What I have learned working on this feature: [If you don't put anything here, you are doing it wrong!]
- (replace me) I learned how to create a pull request template

### CODE REVIEW

File: app/Supplier.php [hunk 1]
```diff
@@ -31,12 +31,12 @@ public static function rules()
     public static function updateRules($id)
     {
         return [
-            'name' => 'sometimes|required|unique:suppliers,name',
+            'name' => 'sometimes|required|string',
```
Comment (AltaraNg) [inline app/Supplier.php:34]: what if you're trying to update with an existing name?

File: package.json [hunk 2]
```diff
@@ -38,7 +38,8 @@
         "ts-jest": "^25.2.0",
         "vue": "^2.5.16",
         "vue-jest": "^3.0.5",
-        "vue-router-back-button": "^1.1.1"
+        "vue-router-back-button": "^1.1.1",
+        "webpack-cli": "^3.3.12"
```
Comment (AltaraNg) [inline package.json:42]: whats the need for these added packages?

File: package.json [hunk 3]
```diff
@@ -53,6 +54,7 @@
         "vue-sweetalert2": "^3.0.3",
         "vue-truncate-filter": "^1.1.7",
         "vue2-datepicker": "^3.5.0",
+        "vue2-filters": "^0.11.0",
```
Comment (AltaraNg) [inline package.json:57]: this

File: resources/assets/js/views/LOG/brand/brands.vue [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    <transition name="fade">
+        <div id="reminder" class="attendance">
+
+            <custom-header :title="'Brands List'"/>
+            <div class="mt-2 mt-lg-3 row attendance-head ">
+                <router-link :to="{name: 'BrandCreate'}">
+                    <button class="btn btn-primary bg-default mt-0 myBtn ">New Brand</button>
+                </router-link>
+            </div>
+<!--            <div class="mt-2 mt-lg-3 row attendance-head attendance-view">-->
```
Comment (AltaraNg) [inline resources/assets/js/views/LOG/brand/brands.vue:11]: lets remove commented code

File: resources/assets/js/views/LOG/brand/brands.vue [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+            }
+        },
+
+        methods: {
+            fetchData() {
+                this.$scrollToTop();
+                this.$LIPS(true);
+                let {page, page_size} = this.$data;
+                get(this.urlToFetchOrders +
+                    `${!!page ? `?page=${page}` : ''}` +
```
Comment (AltaraNg) [inline resources/assets/js/views/LOG/brand/brands.vue:167]: Why not use template string to make it more neat?

File: resources/assets/js/views/LOG/category/categories.vue [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+                    <button class="btn btn-primary bg-default mt-0 myBtn ">New Category</button>
+                </router-link>
+            </div>
+<!--Todo Create filter component -->
+<!--            <div class="mt-2 mt-lg-3 row attendance-head attendance-view">-->
+<!--                <div class="col-4 col-lg" v-for="{name:filter,model} in filters">-->
+<!--                    <div class="row">-->
+
+<!--                        <div class="light-heading mb-1">-->
+<!--                            <span class="d-none d-sm-inline">Select </span>-->
```
Comment (AltaraNg) [inline resources/assets/js/views/LOG/category/categories.vue:17]: let remove commented codes

File: package.json (hunk 2 above)
Comment (darmhoo) [reply package.json:42]: It was a required update for the app to run on my system

Comment (darmhoo) [reply package.json:42]: When i had issues with npm then, I deleted and node module and started again

File: package.json [hunk 7]
```diff
@@ -38,7 +38,8 @@
         "ts-jest": "^25.2.0",
         "vue": "^2.5.16",
         "vue-jest": "^3.0.5",
-        "vue-router-back-button": "^1.1.1"
+        "vue-router-back-button": "^1.1.1",
```
Comment (AltaraNg) [inline package.json:41]: What are these packages for?

File: resources/assets/js/views/LOG/brand/brands.vue [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+                            <button @click="edit(brandItem.id)" class="text-center btn bg-default">Edit</button>
+                        </div>
+                    </div>
+                </div>
+            </div>
+
+            <div v-if="pageParams">
+                <base-pagination
+                    :prev_page_url="pageParams.prev_page_url? pageParams.prev_page_url : '' "
+                    :next_page_url="pageParams.next_page_url? pageParams.next_page_url : ''"
```
Comment (AltaraNg) [inline resources/assets/js/views/LOG/brand/brands.vue:69]: why is there so much binding on pagination?

**Pull request merged**